File: src/backend/nfe/src/model/nfe_detalhe_imposto_icms_ufdest_model.py
from src import db


class NfeDetalheImpostoIcmsUfdestModel(db.Model):
    __tablename__ = 'nfe_detalhe_imposto_icms_ufdest'

    id = db.Column(db.Integer, primary_key=True)
    valor_bc_icms_uf_destino = db.Column(db.Float)
    valor_bc_fcp_uf_destino = db.Column(db.Float)
    percentual_fcp_uf_destino = db.Column(db.Float)
    aliquota_interna_uf_destino = db.Column(db.Float)
    aliquota_interesdatual_uf_envolvidas = db.Column(db.Float)
    percentual_provisorio_partilha_icms = db.Column(db.Float)
    valor_icms_fcp_uf_destino = db.Column(db.Float)
    valor_interestadual_uf_destino = db.Column(db.Float)
    valor_interestadual_uf_remetente = db.Column(db.Float)
    id_nfe_detalhe = db.Column(db.Integer, db.ForeignKey('nfe_detalhe.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_nfe_detalhe = data.get('idNfeDetalhe')
        self.valor_bc_icms_uf_destino = data.get('valorBcIcmsUfDestino')
        self.valor_bc_fcp_uf_destino = data.get('valorBcFcpUfDestino')
        self.percentual_fcp_uf_destino = data.get('percentualFcpUfDestino')
        self.aliquota_interna_uf_destino = data.get('aliquotaInternaUfDestino')
        self.aliquota_interesdatual_uf_envolvidas = data.get('aliquotaInteresdatualUfEnvolvidas')
        self.percentual_provisorio_partilha_icms = data.get('percentualProvisorioPartilhaIcms')
        self.valor_icms_fcp_uf_destino = data.get('valorIcmsFcpUfDestino')
        self.valor_interestadual_uf_destino = data.get('valorInterestadualUfDestino')
        self.valor_interestadual_uf_remetente = data.get('valorInterestadualUfRemetente')

    def serialize(self):
        return {
            'id': self.id,
            'idNfeDetalhe': self.id_nfe_detalhe,
            'valorBcIcmsUfDestino': self.valor_bc_icms_uf_destino,
            'valorBcFcpUfDestino': self.valor_bc_fcp_uf_destino,
            'percentualFcpUfDestino': self.percentual_fcp_uf_destino,
            'aliquotaInternaUfDestino': self.aliquota_interna_uf_destino,
            'aliquotaInteresdatualUfEnvolvidas': self.aliquota_interesdatual_uf_envolvidas,
            'percentualProvisorioPartilhaIcms': self.percentual_provisorio_partilha_icms,
            'valorIcmsFcpUfDestino': self.valor_icms_fcp_uf_destino,
            'valorInterestadualUfDestino': self.valor_interestadual_uf_destino,
            'valorInterestadualUfRemetente': self.valor_interestadual_uf_remetente,
        }

File: frontend/nfe/lib/app/data/model/nfe_detalhe_imposto_icms_ufdest_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';


class NfeDetalheImpostoIcmsUfdestModel {
	int? id;
	int? idNfeDetalhe;
	double? valorBcIcmsUfDestino;
	double? valorBcFcpUfDestino;
	double? percentualFcpUfDestino;
	double? aliquotaInternaUfDestino;
	double? aliquotaInteresdatualUfEnvolvidas;
	double? percentualProvisorioPartilhaIcms;
	double? valorIcmsFcpUfDestino;
	double? valorInterestadualUfDestino;
	double? valorInterestadualUfRemetente;

	NfeDetalheImpostoIcmsUfdestModel({
		this.id,
		this.idNfeDetalhe,
		this.valorBcIcmsUfDestino,
		this.valorBcFcpUfDestino,
		this.percentualFcpUfDestino,
		this.aliquotaInternaUfDestino,
		this.aliquotaInteresdatualUfEnvolvidas,
		this.percentualProvisorioPartilhaIcms,
		this.valorIcmsFcpUfDestino,
		this.valorInterestadualUfDestino,
		this.valorInterestadualUfRemetente,
	});

	static List<String> dbColumns = <String>[
		'id',
		'valor_bc_icms_uf_destino',
		'valor_bc_fcp_uf_destino',
		'percentual_fcp_uf_destino',
		'aliquota_interna_uf_destino',
		'aliquota_interesdatual_uf_envolvidas',
		'percentual_provisorio_partilha_icms',
		'valor_icms_fcp_uf_destino',
		'valor_interestadual_uf_destino',
		'valor_interestadual_uf_remetente',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Valor Bc Icms Uf Destino',
		'Valor Bc Fcp Uf Destino',
		'Percentual Fcp Uf Destino',
		'Aliquota Interna Uf Destino',
		'Aliquota Interesdatual Uf Envolvidas',
		'Percentual Provisorio Partilha Icms',
		'Valor Icms Fcp Uf Destino',
		'Valor Interestadual Uf Destino',
		'Valor Interestadual Uf Remetente',
	];

	NfeDetalheImpostoIcmsUfdestModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idNfeDetalhe = jsonData['idNfeDetalhe'];
		valorBcIcmsUfDestino = jsonData['valorBcIcmsUfDestino']?.toDouble();
		valorBcFcpUfDestino = jsonData['valorBcFcpUfDestino']?.toDouble();
		percentualFcpUfDestino = jsonData['percentualFcpUfDestino']?.toDouble();
		aliquotaInternaUfDestino = jsonData['aliquotaInternaUfDestino']?.toDouble();
		aliquotaInteresdatualUfEnvolvidas = jsonData['aliquotaInteresdatualUfEnvolvidas']?.toDouble();
		percentualProvisorioPartilhaIcms = jsonData['percentualProvisorioPartilhaIcms']?.toDouble();
		valorIcmsFcpUfDestino = jsonData['valorIcmsFcpUfDestino']?.toDouble();
		valorInterestadualUfDestino = jsonData['valorInterestadualUfDestino']?.toDouble();
		valorInterestadualUfRemetente = jsonData['valorInterestadualUfRemetente']?.toDouble();
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idNfeDetalhe'] = idNfeDetalhe != 0 ? idNfeDetalhe : null;
		jsonData['valorBcIcmsUfDestino'] = valorBcIcmsUfDestino;
		jsonData['valorBcFcpUfDestino'] = valorBcFcpUfDestino;
		jsonData['percentualFcpUfDestino'] = percentualFcpUfDestino;
		jsonData['aliquotaInternaUfDestino'] = aliquotaInternaUfDestino;
		jsonData['aliquotaInteresdatualUfEnvolvidas'] = aliquotaInteresdatualUfEnvolvidas;
		jsonData['percentualProvisorioPartilhaIcms'] = percentualProvisorioPartilhaIcms;
		jsonData['valorIcmsFcpUfDestino'] = valorIcmsFcpUfDestino;
		jsonData['valorInterestadualUfDestino'] = valorInterestadualUfDestino;
		jsonData['valorInterestadualUfRemetente'] = valorInterestadualUfRemetente;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idNfeDetalhe = plutoRow.cells['idNfeDetalhe']?.value;
		valorBcIcmsUfDestino = plutoRow.cells['valorBcIcmsUfDestino']?.value?.toDouble();
		valorBcFcpUfDestino = plutoRow.cells['valorBcFcpUfDestino']?.value?.toDouble();
		percentualFcpUfDestino = plutoRow.cells['percentualFcpUfDestino']?.value?.toDouble();
		aliquotaInternaUfDestino = plutoRow.cells['aliquotaInternaUfDestino']?.value?.toDouble();
		aliquotaInteresdatualUfEnvolvidas = plutoRow.cells['aliquotaInteresdatualUfEnvolvidas']?.value?.toDouble();
		percentualProvisorioPartilhaIcms = plutoRow.cells['percentualProvisorioPartilhaIcms']?.value?.toDouble();
		valorIcmsFcpUfDestino = plutoRow.cells['valorIcmsFcpUfDestino']?.value?.toDouble();
		valorInterestadualUfDestino = plutoRow.cells['valorInterestadualUfDestino']?.value?.toDouble();
		valorInterestadualUfRemetente = plutoRow.cells['valorInterestadualUfRemetente']?.value?.toDouble();
	}	

	NfeDetalheImpostoIcmsUfdestModel clone() {
		return NfeDetalheImpostoIcmsUfdestModel(
			id: id,
			idNfeDetalhe: idNfeDetalhe,
			valorBcIcmsUfDestino: valorBcIcmsUfDestino,
			valorBcFcpUfDestino: valorBcFcpUfDestino,
			percentualFcpUfDestino: percentualFcpUfDestino,
			aliquotaInternaUfDestino: aliquotaInternaUfDestino,
			aliquotaInteresdatualUfEnvolvidas: aliquotaInteresdatualUfEnvolvidas,
			percentualProvisorioPartilhaIcms: percentualProvisorioPartilhaIcms,
			valorIcmsFcpUfDestino: valorIcmsFcpUfDestino,
			valorInterestadualUfDestino: valorInterestadualUfDestino,
			valorInterestadualUfRemetente: valorInterestadualUfRemetente,
		);			
	}

	
}